util/kubeconfig/kubeconfig.go

```diff
@@ -96,19 +96,41 @@ func New(clusterName, endpoint string, caCert *x509.Certificate, caKey *rsa.Priv
 	}, nil
 }
 
+// SecretOpt is an option that operates on a v1.Secret type.
+type SecretOpt func(*corev1.Secret)
+
+// WithType returns a SecretOpt that sets the type of the v1.Secret.
+func WithType(st corev1.SecretType) SecretOpt {
```

Comment:
If we consolidate down to a single "type", then we can remove the need for this method and apply that type to all of the Secrets we generate.